app/validators/uri_validator.rb

```diff
@@ -4,12 +4,16 @@
 # Accepts options[:message] and options[:allowed_protocols]
 class UriValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
-    if !(uri = Addressable::URI.parse(value))
+    return if value.nil?
+
+    uri = URI.parse(value)
+
+    if uri.blank?
       record.errors[attribute] << failure_message
-    elsif !allowed_protocols.include?(uri.scheme)
+    elsif allowed_protocols.exclude?(uri.scheme)
       record.errors[attribute] << "is not valid. Make sure it starts with http(s)"
     end
-  rescue Addressable::URI::InvalidURIError
+  rescue URI::InvalidComponentError, URI::InvalidURIError
```

Comment:
Rather than catching two descendent exceptions you could catch just their superclass of `URI::Error`.